src/operations/write_to_file.rs

```diff
@@ -0,0 +1,10 @@
+use std::error::Error;
+use std::fs::File;
+use std::io::Write;
+use std::path::PathBuf;
+
+pub(crate) fn write_to_file(path: &PathBuf, contents: &[u8]) -> Result<(String), Box<dyn Error>> {
```

Comment:
It seems wrong to use Box<dyn Error> here as you could just use std::io::Error. But then you are already using this particular Result type all over the place and I can see how it makes your life easier.